src/effects/Reverb_libSoX.h

```diff
@@ -203,13 +204,13 @@ static void filter_array_init(filter_array_t* p, double rate,
    {
       filter_t* pcomb = &p->comb[i];
       size_t newSize = (size_t)(scale * r * (comb_lengths[i] + stereo_adjust * offset) + .5);
-      filter_t_resize(pcomb, newSize);
+      filter_t_resize(pcomb, std::min(newSize, pcomb->allocSize));
    }
    for (i = 0; i < array_length(allpass_lengths); ++i, offset = -offset)
    {
       filter_t* pallpass = &p->allpass[i];
       size_t newSize = (size_t)(r * (allpass_lengths[i] + stereo_adjust * offset) + .5);
-      filter_t_resize(pallpass, newSize);
+      filter_t_resize(pallpass, std::min(newSize, pallpass->allocSize));
```

Comment:
No.  Something is wrong here.  This change makes the crash go away but changes the calculations.

We thought we allocated sufficient capacity for the largest size required by any parameters.  The fact of the crash tells us our reasoning was wrong somewhere.  Look closer to see where.

Here it is:  in `filter_array_allocate` is this line (twice):
```
   for (i = 0; i < array_length(comb_lengths); ++i, offset = -offset)
```

So as `offset` (a double between 0 and 1) becomes larger, the size of odd-numbered comb and allpass filters becomes *smaller*.

The alternation of offset between positive and negative should be preserved in `filter_array_init` so it computes the same results as before, but should be removed from the allocation function, so that all the filters have sufficient capacity for the size that is computed for any stereo depth value.

(Or really it could alternate between the given offset value and 0 and still make sufficient capacity.)

Please make that change instead.

And make the other changes I suggested too, which are not necessary to fix the crash perhaps, but should be done to be sure the same results are computed.

That is:  `reverb_t` should remember whether `depth` was 0 when `reverb_init` was done, and test that, and not whether `p->out[i]` is null, in `reverb_process`.

Reply:
ok, all done now